### ISSUE TITLE
Add platform specific unit tests

### ISSUE THREAD

**Issue by LasseRosenow** [erlingrj/reactor-uc#123]
Now that we start to have very platform specific implementations for `NetworkChannel`s such as the `CoapUdpIpChannel` for RIOT OS, it makes sense that we add unit tests that are executed on the correct runtime.

I am not yet sure how to integrate the CMAKE stuff to run make based RIOT OS tests, but I think we should find a way.

**Comment by erlingrj** [comment]
Yes, agreed. The first challenge is finding a nice test environment, such as RIOT with the tap interface
